source/include/mqtt_agent.h

```diff
@@ -156,15 +169,15 @@ typedef void (* IncomingPublishCallback_t )( MQTTAgentContext_t * pMqttAgentCont
  * MQTTAgent_Init(), and every API function will accept a pointer to the
  * initalized struct.
  */
-struct MQTTAgentContext
+typedef struct MQTTAgentContext
```

Comment:
Can this definition be moved before the definition of the `IncomingPublishCallback_t` type so that the parameter can take the `MqttAgentContext_t` type?

Reply:
No, since `MQTTAgentContext_t` includes `IncomingPublishCallback_t`